app/sources/HFTclTemplateController.m

```diff
@@ -612,16 +612,21 @@ - (int)runTypeCommand:(enum command)command objc:(int)objc objv:(struct Tcl_Obj
             const char *arg = Tcl_GetStringFromObj(extraArgs[i], NULL);
             if (arg && arg[0] == '-') {
                 Tcl_SetObjResult(_interp, Tcl_ObjPrintf("Unknown option %s", arg));
+                ckfree((char *)extraArgs);
                 return TCL_ERROR;
             }
         }
         if (objc > 2) {
             const char *usage = hexSwitchAllowed ? "[-hex] [label]" : "[label";
             Tcl_WrongNumArgs(_interp, 0, objv, usage);
+            ckfree((char *)extraArgs);
```

Comment:
... so surely at this point extraArgs is NULL, almost guaranteed? Or am I missing something?

Reply:
Because of the ckfree above? but then we have returned error already. Or do i miss something?

Reply:
Oh, never mind I am just very confused. So &extraArgs is not null when Tcl_ParseArgsObjv is called?

Reply:
Mm confusing semantics. I under it as if objc is not zero the array should be freed. But maybe a non-null check would not hurt also